core/src/main/java/org/hisp/dhis/android/core/arch/api/fields/internal/FieldsConverterFactory.kt

```diff
@@ -25,32 +25,17 @@
  *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+package org.hisp.dhis.android.core.arch.api.fields.internal
 
-package org.hisp.dhis.android.core.arch.api.fields.internal;
+import org.hisp.dhis.android.core.arch.api.filters.internal.Which
+import retrofit2.Converter
+import retrofit2.Retrofit
+import java.lang.reflect.Type
 
-import org.hisp.dhis.android.core.arch.api.filters.internal.Which;
-
-import java.lang.annotation.Annotation;
-import java.lang.reflect.Type;
-
-import retrofit2.Converter;
-import retrofit2.Retrofit;
-
-public final class FieldsConverterFactory extends Converter.Factory {
-    public static FieldsConverterFactory create() {
-        return new FieldsConverterFactory();
-    }
-
-    private FieldsConverterFactory() {}
-
-    @Override
-    public Converter<?, String> stringConverter(Type typef,
-            Annotation[] annotations, Retrofit retrofit) {
-        for (Annotation annotation : annotations) {
-            if (annotation instanceof Which) {
-                return new FieldsConverter();
-            }
-        }
-        return null;
-    }
+internal class FieldsConverterFactory : Converter.Factory() {
+    override fun stringConverter(
+        type: Type,
+        annotations: Array<Annotation>,
+        retrofit: Retrofit,
+    ): Converter<*, String>? = annotations.firstOrNull { it is Which }?.let { FieldsConverter() }
```

Comment:
Now only the first annotation is checked for type Which, but previously the array of annotations was looped on until one was of type Which. Is this intended?

Reply:
Hey! `firstOrNull` usually helps to safely retrieve the first element of a collection or sequence, or return null if the collection is empty. But you can also use `firstOrNull` with a predicate to find the first element that matches a given condition. If no such element exists, it returns null.

In this case the condition is `it is Which`, so it loops the annotations until it finds that annotation.